internal/service/athena/capacity_reservation.go

```diff
@@ -0,0 +1,350 @@
+		},
+	}
+}
+
+func (r *resourceCapacityReservation) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	conn := r.Meta().AthenaClient(ctx)
+
+	var plan resourceCapacityReservationModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var input athena.CreateCapacityReservationInput
+	resp.Diagnostics.Append(flex.Expand(ctx, plan, &input)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	input.Tags = getTagsIn(ctx)
+
+	if _, err := conn.CreateCapacityReservation(ctx, &input); err != nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.Athena, create.ErrActionCreating, ResNameCapacityReservation, plan.Name.String(), err),
+			err.Error(),
+		)
+		return
+	}
+
+	out, err := waitCapacityReservationActive(ctx, conn, plan.Name.ValueString(), r.CreateTimeout(ctx, plan.Timeouts))
+	if err != nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.Athena, create.ErrActionWaitingForCreation, ResNameCapacityReservation, plan.Name.String(), err),
+			err.Error(),
+		)
+		return
+	}
+
+	resp.Diagnostics.Append(flex.Flatten(ctx, out, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	plan.ARN = flex.StringValueToFramework(ctx, r.buildARN(ctx, plan.Name.ValueString()))
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, plan)...)
+}
+
+func (r *resourceCapacityReservation) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	conn := r.Meta().AthenaClient(ctx)
+
+	var state resourceCapacityReservationModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	out, err := findCapacityReservationByName(ctx, conn, state.Name.ValueString())
+	if tfresource.NotFound(err) {
+		resp.Diagnostics.Append(fwdiag.NewResourceNotFoundWarningDiagnostic(err))
+		resp.State.RemoveResource(ctx)
+		return
+	}
+	if err != nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.Athena, create.ErrActionReading, ResNameCapacityReservation, state.Name.String(), err),
+			err.Error(),
+		)
+		return
+	}
+
+	resp.Diagnostics.Append(flex.Flatten(ctx, out, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	state.ARN = flex.StringValueToFramework(ctx, r.buildARN(ctx, state.Name.ValueString()))
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+func (r *resourceCapacityReservation) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	conn := r.Meta().AthenaClient(ctx)
+
+	var plan, state resourceCapacityReservationModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if !plan.TargetDPUs.Equal(state.TargetDPUs) {
+		var input athena.UpdateCapacityReservationInput
+		resp.Diagnostics.Append(flex.Expand(ctx, plan, &input, flex.WithFieldNamePrefix("Test"))...)
```

Comment:
`flex.WithFieldNamePrefix("Test")` accidental?

Reply:
Yes, thank you.